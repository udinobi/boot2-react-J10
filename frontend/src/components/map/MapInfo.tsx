import ol from "ol"
import Proj from "ol/proj"

import React from "react"

import styled from "styled-components"

import { num2Text } from "../../lib/utils"

export interface MapInfoState {
    // The coordinate system OpenLayers uses by default for the map view is (Web Mercator EPSG: 3857).
    coord: ol.Coordinate
    zoom: number
}

const CoordInfo = styled.span`
    padding-left: 38px;
`

const MapInfo = styled.div`
    @media (max-width: 576px) {
        font-size: 0.9rem;
    }
    padding-bottom: 6px;
`
                                                   // 'EPSG:3857' -> 'EPSG:4326'
const toLonLat = (coord: [number, number]) => Proj.toLonLat([ coord[0], coord[1] ])

export default (props: MapInfoState) => {
    const coord = toLonLat(props.coord)
    return <MapInfo>
        zoom level ({props.zoom})
        <CoordInfo>
            [ {num2Text(coord[1], 8)}, {"\u00a0"}{num2Text(coord[0], 9)} ]
        </CoordInfo>
    </MapInfo>
}
