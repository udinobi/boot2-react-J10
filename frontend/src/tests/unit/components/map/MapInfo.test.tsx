import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import MapInfo, { MapInfoState } from "../../../../components/map/MapInfo"

describe("--> Snapshot for Components > map > 'MapInfo'", () => {
    let props: MapInfoState

    beforeEach(() => {
        props = {
            coord: [ 0, 0 ],
            zoom: 11
        }
    })

    it("matches", () => {
        const wrapper = shallow(<MapInfo {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
