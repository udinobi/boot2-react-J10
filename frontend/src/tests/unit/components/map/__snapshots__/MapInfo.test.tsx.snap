// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > map > 'MapInfo' matches 1`] = `
<styled.div>
  zoom level (
  11
  )
  <styled.span>
    [ 
    0
    , 
    Â 
    0
     ]
  </styled.span>
</styled.div>
`;
